### PR TITLE
Settings: Add Markdown setting for JP sites

### LINKED ISSUE #16884: Add markdown settings in for Jetpack sites only
If I remember correctly, we removed markdown all across Calypso. There was an effort to remove markdown settings in Jetpack sites as well so we decided not to add markdown settings for Jetpack sites. However, this turned out to be somewhat confusing for Jetpack users.

We currently have markdown options in Jetpack and I think we should mirror them in Calypso. Perhaps we can reuse some of the work @ryelle did here: https://github.com/Automattic/wp-calypso/pull/10785

Here are what the options look like in Jetpack:
image
image

I propose we do the same thing within the Calypso settings by adding the first option exactly as it is in Jetpack to the Composing settings group within the Writing settings tab. We should also add the option to use it in comments in the Comments group of the Discussion settings tab.

CC @oskosk @tyxla

### PR DESCRIPTION
**Pull request by ockham**
Add settings to enable Markdown editing on Jetpack sites, mimicking Jetpack's wp-admin UI -- see screenshots in #16884 (with one exception -- the `InfoPopover` next to 'Write posts or pages in plain-text Markdown syntax' just felt like terrible UX, which is why I opted for the 'Learn more' link instead).

### Before:

`/settings/writing/<JPsite>`, 'Composing' section:

image

`/settings/discussion/<JPsite>`, bottom of 'Comments' section:

image

### After:

`/settings/writing/<JPsite>`, 'Composing' section:

image

`/settings/discussion/<JPsite>`, bottom of 'Comments' section:

image

### To test

* Verify that these settings are only present for JP sites, but not for WP.com ones.
* Verify that the 'Enable Markdown for comments.' toggle is not shown (by default), unless the 'Write posts or pages in plain-text Markdown syntax' toggle is enabled.
* Verify that flipping the 'Write posts or pages in plain-text Markdown syntax' toggle works (no errors, toggle state persisted after a reload), and causes the 'Enable Markdown for comments.' toggle to be shown.
* Verify that the 'Enable Markdown for comments.' works.
* Verify that once enabled, you can actually use Markdown to write posts and comments, respectively.

Fixes #16884.

### CODE REVIEW

File: client/my-sites/site-settings/composing/markdown.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	};
+
+	render() {
+		const { disabled, siteId, translate } = this.props;
+
+		return (
+			<FormFieldset className="composing__markdown has-divider is-bottom-only">
+				<JetpackModuleToggle
+					disabled={ disabled }
+					label={ translate( 'Write posts or pages in plain-text Markdown syntax' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/markdown.jsx]: Hi! I've found a possible matching string that has already been translated **11** times:
`translate( 'Write posts or pages in plain text Markdown syntax.' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/composing/markdown.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	};
+
+	render() {
+		const { disabled, siteId, translate } = this.props;
+
+		return (
+			<FormFieldset className="composing__markdown has-divider">
+				<JetpackModuleToggle
+					disabled={ disabled }
+					label={ translate( 'Write posts or pages in plain-text Markdown syntax' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/markdown.jsx]: Hi! I've found a possible matching string that has already been translated **11** times:
`translate( 'Write posts or pages in plain text Markdown syntax.' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by designsimply** [comment]
I noticed that the comments markdown option in Settings > Discussion has a "Learn more" link next to it in the screenshot at #16884 but the `add/jetpack-markdown-settings` branch does not have one that. Did you mean to leave it out?

28937068-eaafac48-7846-11e7-8a34-7cce42e78a5b

screen shot 2018-03-06 at tue mar 6 3 06 58 pm
<sup>Seen at https://calypso.live/settings/discussion/madefortesting.com?branch=add/jetpack-markdown-settings using Firefox 58.0.2 on macOS 10.13.3.</sup>

**Comment by designsimply** [comment]
Editing posts with Markdown in Calypso is a bit problematic. See [25s](https://cloudup.com/cA96eqPLDPT) from #20608 and this [3m38s](https://cloudup.com/cRgtzECd33r) example.

> Markdown support has never been addressed in REST API contexts and therefore also not in the Calypso editor.

Source: https://github.com/Automattic/wp-calypso/issues/2373#issuecomment-278138284

> It's very frustrating using markdown to create footnotes as they don't work.

Source: p2-p4k3M4-2uO (internal ref)

It's possible you may want to consider holding off enabling markdown for Jetpack sites in Calypso until Calypso has support for Markdown. cc @MichaelArestad

**Comment by designsimply** [comment]
I checked the following with Simple, Atomic, and Jetpack sites using https://calypso.live/?branch=add/jetpack-markdown-settings

1. Markdown settings are present for Jetpack and Atomic, not for Simple.
1. Markdown for comments is not avail. on Jetpack/Atomic unless markdown for posts is enabled.
1. Enabling/disabling each setting works as expected (refreshing the page shows the expected state).
1. Disabling markdown for posts also disables markdown for comments automatically.
1. It is possible to start a post in Calypso with Markdown but not to continue editing it using Markdown because it is converted to HTML. ([see previous comment](https://github.com/Automattic/wp-calypso/pull/22999#issuecomment-370954874))
1. It is possible to submit a comment using Markdown but you cannot edit comment Markdown later because it is converted to HTML. For comments, this is the current behavior in WP Admin as well as Calypso.

**Comment by tyxla** [comment]
Seems like we're opting to remove this one instead - https://github.com/Automattic/wp-calypso/issues/16884#issuecomment-370870635

**Comment by ockham** [comment]
Okay, closing 😅 Sorry for the noise everybody -- especially @designsimply, appreciate your testing! 🙂 
I'll file a PR to remove Markdown settings on the JP side instead.

**Comment by ockham** [comment]
> I'll file a PR to remove Markdown settings on the JP side instead.

https://github.com/Automattic/jetpack/pull/9015

File: client/my-sites/site-settings/composing/markdown.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import FormFieldset from 'components/forms/form-fieldset';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import ExternalLink from 'components/external-link';
+import JetpackModuleToggle from '../jetpack-module-toggle';
+
+const Markdown = ( { disabled, siteId, translate } ) => (
+	<FormFieldset className="composing__markdown has-divider is-top-only">
+		<JetpackModuleToggle
+			disabled={ disabled }
+			label={ translate( 'Write posts or pages in plain-text Markdown syntax' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/composing/markdown.jsx]: Hi! I've found a possible matching string that has already been translated **11** times:
`translate( 'Write posts or pages in plain text Markdown syntax.' )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request closed without merging**